code/game/gamemodes/miniantags/guardian/types/bomb.dm

```diff
@@ -37,38 +37,40 @@
 /obj/item/guardian_bomb/proc/disguise(var/obj/A)
 	A.forceMove(src)
 	stored_obj = A
+	opacity = A.opacity
 	anchored = A.anchored
 	density = A.density
 	appearance = A.appearance
-	spawn(600)
-		if(src)
-			stored_obj.loc = get_turf(loc)
-			if(spawner)
-				to_chat(spawner, "<span class='danger'>Failure! Your trap on [stored_obj] didn't catch anyone this time.</span>")
-			qdel(src)
+	dir = A.dir
+	addtimer(CALLBACK(src, .proc/disable), 600)
+
+/obj/item/guardian_bomb/proc/disable()
+	stored_obj.forceMove(get_turf(src))
+	to_chat(spawner, "<span class='danger'>Failure! Your trap on [stored_obj] didn't catch anyone this time.</span>")
```

Comment:
I do believe this needs that `if(spawner)` or to_chat whines in the runtime logs

Reply:
whoops, fixed